lib/Frontend/ModuleInterfaceLoader.cpp

```diff
@@ -391,12 +391,17 @@ class ModuleInterfaceLoaderImpl {
     // anyways.
     H = hash_combine(H, interfacePath);
 
-    // Include the target CPU architecture. In practice, .swiftinterface files
-    // will be in architecture-specific subdirectories and would have
-    // architecture-specific pieces #if'd out. However, it doesn't hurt to
+    // Include the normalized target triple. In practice, .swiftinterface files
+    // will be in target-specific subdirectories and would have
+    // target-specific pieces #if'd out. However, it doesn't hurt to
     // include it, and it guards against mistakenly reusing cached modules
-    // across architectures.
-    H = hash_combine(H, SubInvocation.getLangOptions().Target.getArchName());
+    // across targets. Note that we this normalization explicitly doesn't
+    // include the minimum deployment target (e.g. the '12.0' in 'ios12.0').
+    // Different modules with different minimum deployment targets are fine to
+    // reuse the same .swiftmodule.
```

Comment:
Hm. It's fine for *clients* with different deployment targets, but if the same swiftinterface is somehow compiled with a different deployment target it may behave differently. That won't happen in real life because the deployment target is always in the swiftinterface, though.

Reply:
That's a good point, I'll reword it.

Reply:
It's a little funny because it's "the sub-invocation's target", but we haven't actually read the options for the sub-invocation's options yet, so it's still coming from the client for now.